### DISCUSSION TITLE
Fatal Exception: java.lang.NullPointerException: file must not be null

### DISCUSSION THREAD

**Discussion by peter100u** [Tencent/libpag#1944] [General]
最近老是出现这个错误，
Fatal Exception: java.lang.NullPointerException: file must not be null
       at io.liuliu.inyu_common.ex.c.onLoad(:6)
       at org.libpag.PAGImageView.a(SourceFile:2)
       at org.libpag.PAGImageView.a()
       at org.libpag.f.run(:8)

奇怪的是，这个file理论上不可能是null的呢，我确定传入的url，不会是null或者"",  有没有一种是pag底层创建的file失败的了，导致底层出现了错误，无法重现，但是firebase上一直有个错误，从使用setPathAsync加入一个url开始

**Comment by peter100u** [comment]
我看到pag中的源码，如下
public boolean setPath(String path) {
        PAGFile pagFile;
        if (path != null && path.startsWith("assets://")) {
            pagFile = PAGFile.Load(getContext().getAssets(), path.substring(9));
        } else {
            pagFile = PAGFile.Load(path);
        }
        setComposition(pagFile);
        filePath = path;
        return pagFile != null;
    }

在path为空的时候，assets情况下进行了容错，但是url情况下，直接PAGFile.load里有直接crash, 希望能容错下

**Comment by domchen** [comment]
> 我看到pag中的源码，如下 public boolean setPath(String path) { PAGFile pagFile; if (path != null && path.startsWith("assets://")) { pagFile = PAGFile.Load(getContext().getAssets(), path.substring(9)); } else { pagFile = PAGFile.Load(path); } setComposition(pagFile); filePath = path; return pagFile != null; }
> 
> 在path为空的时候，assets情况下进行了容错，但是url情况下，直接PAGFile.load里有直接crash, 希望能容错下

URL是在 PAGFile.Load() 里统一处理的，这里代码没有什么问题。报错的内容是 Android 系统输出的，报错内容里的 file 并不是指 PAGFile。先还原一下你的堆栈信息，里面有很多符号被混淆了看不出来是什么函数。然后尝试构造一个复现的demo上传上来定位会比较快。

**Comment by CodeJhF** [comment]
这条我们在当时的回复里已经给出结论，这里就不重复贴了，先关闭本条。
如果按当时的结论排查后仍有问题，欢迎回复补充最小复现 demo 与版本号（4.4 / 4.5），我们会重新看。（stale-close）
